velox/exec/tests/OperatorTraceTest.cpp

```diff
@@ -727,6 +728,252 @@ TEST_F(OperatorTraceTest, filterProject) {
+    ASSERT_TRUE(fs->exists(summaryFilePath));
+    ASSERT_TRUE(fs->exists(splitFilePath));
+
+    traceDirs.push_back(opTraceDir);
+  }
+
+  const auto reader = exec::trace::OperatorTraceSplitReader(
+      traceDirs, memory::MemoryManager::getInstance()->tracePool());
+  auto actualSplits = reader.read();
+  std::unordered_set<std::string> splitStrs;
+  std::transform(
+      splits.begin(),
+      splits.end(),
+      std::inserter(splitStrs, splitStrs.begin()),
+      [](const auto& s) { return s->toString(); });
+  ASSERT_EQ(actualSplits.size(), splits.size());
+  for (int i = 0; i < numSplits; ++i) {
+    folly::dynamic splitInfoObj = folly::parseJson(actualSplits[i]);
+    const auto actualSplit = exec::Split{
+        std::const_pointer_cast<connector::hive::HiveConnectorSplit>(
+            ISerializable::deserialize<connector::hive::HiveConnectorSplit>(
+                splitInfoObj))};
+    ASSERT_FALSE(actualSplit.hasGroup());
+    ASSERT_TRUE(actualSplit.hasConnectorSplit());
+    const auto actualConnectorSplit = actualSplit.connectorSplit;
+    ASSERT_EQ(splitStrs.count(actualConnectorSplit->toString()), 1);
+  }
+}
+
+TEST_F(OperatorTraceTest, traceSplitPartial) {
+  constexpr auto numSplits = 3;
+  const auto vectors = makeVectors(2, 100);
+  const auto testDir = TempDirectoryPath::create();
+  const auto traceRoot = fmt::format("{}/{}", testDir->getPath(), "traceRoot");
+  const auto fs = filesystems::getFileSystem(testDir->getPath(), nullptr);
+  std::vector<std::shared_ptr<TempFilePath>> splitFiles;
+  for (int i = 0; i < numSplits; ++i) {
+    auto filePath = TempFilePath::create();
+    writeToFile(filePath->getPath(), vectors);
+    splitFiles.push_back(std::move(filePath));
+  }
+  auto splits = makeHiveConnectorSplits(splitFiles);
+
+  auto traceDirPath = TempDirectoryPath::create();
+  auto plan = PlanBuilder().tableScan(dataType_).planNode();
+  std::shared_ptr<Task> task;
+  AssertQueryBuilder(plan)
+      .maxDrivers(3)
+      .config(core::QueryConfig::kQueryTraceEnabled, true)
+      .config(core::QueryConfig::kQueryTraceDir, traceDirPath->getPath())
+      .config(core::QueryConfig::kQueryTraceTaskRegExp, ".*")
+      .config(core::QueryConfig::kQueryTraceNodeIds, "0")
+      .splits(splits)
+      .copyResults(pool(), task);
+
+  const auto taskTraceDir =
+      getTaskTraceDirectory(traceDirPath->getPath(), *task);
+  std::vector<std::string> traceDirs;
+  for (int i = 0; i < 3; ++i) {
+    const auto opTraceDir = getOpTraceDirectory(
+        taskTraceDir,
+        /*planNodeId=*/"0",
+        /*pipelineId=*/0,
+        /*driverId=*/i);
+    const auto summaryFilePath = getOpTraceSummaryFilePath(opTraceDir);
+    const auto splitFilePath = getOpTraceSplitFilePath(opTraceDir);
+    ASSERT_TRUE(fs->exists(summaryFilePath));
+    ASSERT_TRUE(fs->exists(splitFilePath));
+
+    traceDirs.push_back(opTraceDir);
+  }
+
+  // Append a partial split to the split info file.
+  const std::string split = "deadbeaf";
```

Comment:
Why call this? Can you change a better name?

Reply:
rename to split123